src/main/java/com/epam/aidial/cfg/service/transfer/CoreConfigVersionAutoDetectService.java

```diff
@@ -0,0 +1,99 @@
+package com.epam.aidial.cfg.service.transfer;
+
+import com.epam.aidial.cfg.client.CoreConfigClient;
+import com.epam.aidial.cfg.configuration.CoreConfigVersionProperties;
+import com.epam.aidial.cfg.configuration.logging.LogExecution;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Service
+@Slf4j
+@LogExecution
+@RequiredArgsConstructor
+public class CoreConfigVersionAutoDetectService {
+
+    private final CoreConfigClient coreConfigClient;
+    private final CoreConfigVersionProperties coreConfigVersionProperties;
+    private final AtomicReference<String> cachedVersion = new AtomicReference<>();
```

Comment:
add cache expiration. it'd be useful to have ability to switch in new core version w/o restart